lib/logstash/outputs/elasticsearch/common_configs.rb

```diff
@@ -81,7 +81,7 @@ def self.included(mod)
       # If you specify a number larger than the batch size of your pipeline it will have no effect,
       # save for the case where a filter increases the size of an inflight batch by outputting
       # events.
-      mod.config :flush_size, :validate => :number, :default => 500
+      mod.config :flush_size, :validate => :number, :default => 99999999, :deprecate => "This setting is no longer necessary as we now try to restrict bulk requests to sane sizes. See the 'Batch Sizes' section of the docs. If you think you still need to restrict payloads based on the number, not size, of events, please open a ticket."
```

Comment:
can we default to nil? A value of 9999999 looks weird (will show up in the docs) and confuses me, and may confuse users.